api/s2n.h

```diff
@@ -596,6 +596,258 @@ extern int s2n_cert_get_utf8_string_from_extension_data_length(const uint8_t *ex
+ * @param secret The pre-shared secret in raw bytes format.
+ * @param secret_size The length of the pre-shared secret being set.
+ */
+S2N_API 
+int s2n_psk_set_secret(struct s2n_psk *psk, const uint8_t *secret, uint16_t secret_size);
+
+/**
+ * Sets the pre-shared key (PSK) HMAC algorithm for a given s2n_psk object. 
+ * The supported pre-shared key HMAC algorithms are `SHA256` and `SHA384` 
+ * as listed in the enum `s2n_psk_hmac` above.
+ * 
+ * @param psk A pointer to the s2n_psk object to be updated with the pre-shared key HMAC algorithm.
+ * @param hmac The pre-shared key HMAC algorithm being set.  
+ */
+S2N_API 
+int s2n_psk_set_hmac(struct s2n_psk *psk, s2n_psk_hmac hmac);
+
+/**
+ * Appends a pre-shared key (PSK) object to the list of pre-shared keys supported by the s2n connection. 
+ * If a pre-shared key with a duplicate identity is found, an error is returned and the pre-shared key is not added to the list.
+ * 
+ * @param conn A pointer to the s2n_connection object that contains the list of pre-shared keys supported.
+ * @param psk A pointer to the `s2n_psk` object to be appended to the list of pre-shared keys on the s2n connection.
+ */
+S2N_API 
+int s2n_connection_append_psk(struct s2n_connection *conn, struct s2n_psk *psk);
+
+/**
+ * The list of pre-shared key modes supported by s2n-tls for TLS versions >= TLS1.3.
+ * Currently s2n-tls supports two modes - `S2N_PSK_MODE_RESUMPTION` which represents the pre-shared keys established 
+ * using the previous connection via session resumption, and `S2N_PSK_MODE_EXTERNAL` which represents pre-shared keys 
+ * established out-of-band/externally using a secure mutually agreed upon mechanism.  
+ *
+ */ 
+typedef enum {
+    S2N_PSK_MODE_RESUMPTION,
+    S2N_PSK_MODE_EXTERNAL 
+} s2n_psk_mode;
+
+/**
+ * Sets the pre-shared key mode on the s2n config object. 
+ * The supported pre-shared key modes are listed in the enum `s2n_psk_mode` above. 
+ * 
+ * @param config A pointer to the s2n_config object being updated.
+ * @param mode The pre-shared key mode to be set. The supported psk modes are listed in the enum `s2n_psk_mode` above. 
+ */
+S2N_API 
+int s2n_config_set_psk_mode(struct s2n_config *config, s2n_psk_mode mode);
+
+/**
+ * Sets the pre-shared key mode on the s2n connection object.
+ * The supported pre-shared key modes are listed in the enum `s2n_psk_mode` above. 
+ * This API overrides the pre-shared key mode for the connection and will continue to use the existing 
+ * pre-shared key mode even when a new config is set for the connection.
+ * 
+ * @param conn A pointer to the s2n_connection object being updated.
+ * @param mode The pre-shared key mode to be set.
+ */
+S2N_API 
+int s2n_connection_set_psk_mode(struct s2n_connection *conn, s2n_psk_mode mode);
+
+/**
+ * Gets the negotiated pre-shared key identity length from the s2n connection object. 
+ * If the negotiated pre-shared key does not exist, the value `0` is returned.  
+ * 
```

Comment:
What does "negotiated" mean here? Can we be more descriptive about what this is?